### PR TITLE
Allow the fontweight for the title to be customized

### PR DESCRIPTION
**Pull request by dylancom**
## Description

Currently you can customize the fontFamily and fontSize but not the fontWeight (of the header title).
This PR let's you also change the fontWeight of the header title.

### CODE REVIEW

**Comment by WoLewicki** [comment]
Thanks for your contribution! 
1. In the PR template, there are all the places where the change/addition of a property should be added. It looks like you've missed the READMEs of `native-stack` and `createNativeStackNavigator`.
2. Can you add this change to Android too? It should be rather simple. You can check here: https://github.com/software-mansion/react-native-screens/blob/master/android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java#L236 and extend the `if` to check for both `weight` and `FontFamily`, only `FontFamily` (current), only `weight`.

If you need any help, please comment here.

**Comment by dylancom** [comment]
@WoLewicki I tried Android but it had no effect, see: https://github.com/facebook/react-native/issues/26193.
Seems that if you want for e.g. a bold font, you need to specify a bold font family.
I updated the READMEs, improved the iOS code and added an example.

File: README.md [hunk 1]
```diff
@@ -190,6 +190,10 @@ Customize the font family to be used for the title.
 
 Customize the size of the font to be used for the title.
 
+#### `titleFontWeight`
```
Comment (WoLewicki) [inline README.md:193]: If it is not available on Android, please add the `(iOS only)` in all READMEs and other files with this.

File: TestsExample/App.js [hunk 2]
```diff
@@ -20,10 +20,11 @@ import Test649 from './src/Test649';
...
 import Test702 from './src/Test702';
+import Test706 from './src/Test706';
 import Test713 from './src/Test713';
 
 enableScreens();
 
 export default function App() {
-  return <Test42 />;
-};
+  return <Test706 />;
```
Comment (WoLewicki) [inline TestsExample/App.js:29]: Please don't change the default test file.

File: TestsExample/ios/Podfile.lock [hunk 3]
```diff
@@ -514,4 +514,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 123196e881422a7a7f3a9c228d6113e8de498360
 
-COCOAPODS: 1.9.3
+COCOAPODS: 1.10.0
```
Comment (WoLewicki) [inline TestsExample/ios/Podfile.lock:517]: We shouldn't mix normal changes with such ones in the same PR. Please change it back to what it was.

File: ios/RNSScreenStackHeaderConfig.h [hunk 4]
```diff
@@ -17,6 +17,7 @@ typedef NS_ENUM(NSInteger, RNSStatusBarStyle) {
 @property (nonatomic, retain) NSString *title;
 @property (nonatomic, retain) NSString *titleFontFamily;
 @property (nonatomic, retain) NSNumber *titleFontSize;
+@property (nonatomic, retain) NSString *titleFontWeight;
```
Comment (WoLewicki) [inline ios/RNSScreenStackHeaderConfig.h:20]: I think we could add the same but for `largeTitle`, the changes should look pretty similar to the normal title.

**Review by WoLewicki — CHANGES_REQUESTED** [review]
Looks good! Please apply the changes requested and this should be ready to merge then!

**Comment by dylancom** [comment]
@WoLewicki I applied the requested changes.

File: ios/RNSScreenStackHeaderConfig.m [hunk 5]
```diff
@@ -145,16 +145,18 @@ + (void)setAnimatedConfig:(UIViewController *)vc withConfig:(RNSScreenStackHeade
...
       if (config.titleColor) {
         attrs[NSForegroundColorAttributeName] = config.titleColor;
       }
 
+      NSString *family = config.titleFontFamily ?: nil;
       NSNumber *size = config.titleFontSize ?: @17;
-      if (config.titleFontFamily) {
-        attrs[NSFontAttributeName] = [RCTFont updateFont:nil withFamily:config.titleFontFamily size:size weight:nil style:nil variant:nil scaleMultiplier:1.0];
+      NSString *weight = config.titleFontWeight ?: nil;
+      if (config.titleFontFamily || config.titleFontWeight) {
```
Comment (WoLewicki) [inline ios/RNSScreenStackHeaderConfig.m:158]:
```suggestion
      if (family || weight) {
```

File: ios/RNSScreenStackHeaderConfig.m [hunk 6]
```diff
@@ -163,14 +165,16 @@ + (void)setAnimatedConfig:(UIViewController *)vc withConfig:(RNSScreenStackHeade
...
         NSMutableDictionary *largeAttrs = [NSMutableDictionary new];
         if (config.largeTitleColor || config.titleColor) {
           largeAttrs[NSForegroundColorAttributeName] = config.largeTitleColor ? config.largeTitleColor : config.titleColor;
         }
+        NSString *largeFamily = config.largeTitleFontFamily ?: nil;
         NSNumber *largeSize = config.largeTitleFontSize ?: @34;
-        if (config.largeTitleFontFamily) {
-          largeAttrs[NSFontAttributeName] = [RCTFont updateFont:nil withFamily:config.largeTitleFontFamily size:largeSize weight:nil style:nil variant:nil scaleMultiplier:1.0];
+        NSNumber *largeWeight = config.largeTitleFontWeight ?: nil;
+        if (config.largeTitleFontFamily || config.largeTitleFontWeight) {
```
Comment (WoLewicki) [inline ios/RNSScreenStackHeaderConfig.m:176]:
```suggestion
        if (largeFamily || largeWeight) {
```

File: ios/RNSScreenStackHeaderConfig.m [hunk 7]
```diff
@@ -291,32 +295,36 @@ + (UINavigationBarAppearance*)buildAppearance:(UIViewController *)vc withConfig:
...
     if (config.titleColor) {
       attrs[NSForegroundColorAttributeName] = config.titleColor;
     }
 
+    NSString *family = config.titleFontFamily ?: nil;
     NSNumber *size = config.titleFontSize ?: @17;
-    if (config.titleFontFamily) {
-      attrs[NSFontAttributeName] = [RCTFont updateFont:nil withFamily:config.titleFontFamily size:size weight:nil style:nil variant:nil scaleMultiplier:1.0];
+    NSString *weight = config.titleFontWeight ?: nil;
+    if (config.titleFontFamily || config.titleFontWeight) {
```
Comment (WoLewicki) [inline ios/RNSScreenStackHeaderConfig.m:308]: same as above

File: ios/RNSScreenStackHeaderConfig.m [hunk 8]
```diff
@@ -291,32 +295,36 @@ + (UINavigationBarAppearance*)buildAppearance:(UIViewController *)vc withConfig:
...
     if (config.largeTitleColor || config.titleColor) {
       largeAttrs[NSForegroundColorAttributeName] = config.largeTitleColor ? config.largeTitleColor : config.titleColor;
     }
 
+    NSString *largeFamily = config.largeTitleFontFamily ?: nil;
     NSNumber *largeSize = config.largeTitleFontSize ?: @34;
-    if (config.largeTitleFontFamily) {
-      largeAttrs[NSFontAttributeName] = [RCTFont updateFont:nil withFamily:config.largeTitleFontFamily size:largeSize weight:nil style:nil variant:nil scaleMultiplier:1.0];
+    NSString *largeWeight = config.largeTitleFontWeight ?: nil;
+    if (config.largeTitleFontFamily || config.largeTitleFontWeight) {
```
Comment (WoLewicki) [inline ios/RNSScreenStackHeaderConfig.m:326]: same as above

File: src/index.d.ts [hunk 9]
```diff
@@ -239,6 +244,10 @@ declare module 'react-native-screens' {
      * @description Customize the size of the font to be used for the title.
      */
     titleFontSize?: number;
+    /**
+     * @description Customize the weight of the font to be used for the title.
+     */
```
Comment (WoLewicki) [inline src/index.d.ts:249]:
```suggestion
    /**
     * @host (iOS only)
     * @description Customize the weight of the font to be used for the title.
     */
```

File: src/native-stack/types.tsx [hunk 10]
```diff
@@ -194,6 +195,7 @@ export type NativeStackNavigationOptions = {
   headerTitleStyle?: {
     fontFamily?: string;
     fontSize?: number;
+    fontWeight?: string;
```
Comment (WoLewicki) [inline src/native-stack/types.tsx:198]: Please add this prop to the description above with a note (iOS only)

File: src/native-stack/types.tsx [hunk 11]
```diff
@@ -262,9 +264,7 @@ export type NativeStackNavigationOptions = {
   title?: string;
 };
 
-export type NativeStackNavigatorProps = DefaultNavigatorOptions<
-  NativeStackNavigationOptions
-> &
+export type NativeStackNavigatorProps = DefaultNavigatorOptions<NativeStackNavigationOptions> &
```
Comment (WoLewicki) [inline src/native-stack/types.tsx:267]: Please revert it

**Review by WoLewicki — CHANGES_REQUESTED** [review]
Some minor tweaks.

**Comment by dylancom** [comment]
@WoLewicki Tweaked it.

File: src/native-stack/types.tsx [hunk 12]
```diff
@@ -189,11 +190,13 @@ export type NativeStackNavigationOptions = {
    * Style object for header title. Supported properties:
    * - fontFamily
    * - fontSize
+   * - fontWeight
```
Comment (WoLewicki) [inline src/native-stack/types.tsx:193]:
```suggestion
   * - fontWeight (iOS only)
```

**Comment by WoLewicki** [comment]
Ok it looks like it is ready to be merged :tada: Thanks for your contribution!

**Review by WoLewicki — APPROVED** [review]

**Pull request merged**